app/assets/javascripts/analytics.js.erb

```diff
@@ -2,112 +2,87 @@
+      // this is because the click event target is actually the
+      // internal element, e.g. an <h3> tag wrapped in <a> tags
+      // will register a click target of <h3> not <a> so the
+      // 'href' property will be undefined and we must check its
+      // parents to acquire the href.
+      if(!href) {
+        href = $target.parents('a').attr('href');
+      }
+
+      // Send separate events for internal and external link clicks
+      var regexp = /^https?:/;
+      if(regexp.test(href)) {
+        _gaq.push(['_trackEvent', 'external_link', 'click', href, 1]);
+      } else {
+        _gaq.push(['_trackEvent', 'internal_link', 'click', href, 1]);
+      }
+    });
+
+    // Fire an event whenever a user marks a lesson completed on the lessons index page
+    // else whenever a student marks a lesson not completed on the lessons index page
+    $('[id^=section-lessons__]').on('click', function(e) {
+      if ($(this).children(":first").data('method') === 'post') {
+        _gaq.push(['_trackEvent', 'lesson_completion', 'mark_completed', $(this).children(":first")[0].href, 1]);
+      } else {
+        _gaq.push(['_trackEvent', 'lesson_completion', 'mark_not_completed', $(this).children(":first")[0].href, 1]);
+      }
+    });
+
+    //see_lesson_buttons.html.erb for lesson button analytics
+
+    // Fire an event when a user clicks on the gitter chat link in the individual lesson page
+    $('#gitter-button').on('click', function(e) {
+      _gaq.push(['_trackEvent', 'gitter_chat', 'click_gitter_chat_link', 'lesson_page', 1]);
+    });
+
+    // Fire an event when a user clicks on the floating chat button
+    $('.chat-floating-btn').click(function(e) {
+      _gaq.push(['_trackEvent', 'gitter_chat', 'click_gitter_chat_floating_button', 'global', 1]);
+    });
+
+    // Fire an event when a user clicks on the gitter chat link in the navbar
+    $('.gitter-chat-link').on('click', function(e) {
+      _gaq.push(['_trackEvent', 'gitter_chat', 'click_gitter_chat_link', 'navbar', 1]);
+    });
+
+    // Fire an event whenever someone tries to sign in the standard way
+    $('#login-button').on('click', function(e) {
+      _gaq.push(['_trackEvent', 'signin_buttons', 'click_normal_signin_button', 'signin_button', 1]);
+    })
+
+    // Fire an event whenever someone tries to sign in with Github by clicking that Github button
+    $('.devise-sign-in').on('click', '.btn-github', function(e) {
+      _gaq.push(['_trackEvent', 'signin_buttons', 'click_github_signin_button', 'github_signin_button', 1]);
+    })
+
+    // Fire an event whenever someone tries to sign UP with Github by clicking that Github button
+    $('.button--github').on('click', function(e){
+      _gaq.push(['_trackEvent', 'signup_buttons', 'click_github_signup button', 'github_signup_button', 1]);
+    })
+
+    // Fire an event whenever someone clicks the normal sign up button (e.g. not step 2 of the github flow)
+    $('.sign-up-button').on('click', function(e){
```

Comment:
This ones got a unique value attribute we could use to select it as well.
`[value="Sign up"]`